src/libcore/convert.rs

```diff
@@ -45,6 +69,12 @@ use marker::Sized;
 /// let s = "hello".to_string();
 /// is_hello(s);
 /// ```
+///
+/// # Generic Impls
+///
+/// - `AsRef` auto-dereference if the inner type is a reference or a mutable
+/// reference
```

Comment:
Could you elaborate on what this is explaining? Is this referring to how `foo.bar()` will work regardless if `foo` has type `&mut &&mut Foo`?